src/StoreApi/Routes/Checkout.php

```diff
@@ -164,6 +164,31 @@ protected function get_route_post_response( \WP_REST_Request $request ) {
 		// Check order is still valid.
 		$order_controller->validate_order_before_payment( $order_object );
 
+		// Create a new user account as necessary.
+		if ( ! is_user_logged_in() && ( WC()->checkout()->is_registration_required() || ! empty( $request['create_account'] ) ) ) {
+			$username    = '';
+			$password    = '';
+			$customer_id = wc_create_new_customer(
+				$order_object->get_billing_email(),
+				$username,
+				$password,
+				array(
+					'first_name' => $order_object->get_billing_first_name(),
+					'last_name'  => $order_object->get_billing_last_name(),
+				)
+			);
+
+			if ( is_wp_error( $customer_id ) ) {
+				return $this->get_route_error_response(
+					400,
```

Comment:
I see we have a few deliberate `500` errors triggered in here – is that by design? I would expect 500 errors only in exceptional, non-expected circumstances, and all other errors should use a value in the 4xx range. I've gone with 400 here for now, but I'm keen to understand our approach for API error codes in blocks.

Reply:
Look up standard HTTP error codes. 4xx are client errors. 5xx are server errors. If the request fails, try to derive why from the WP error.

Reply:
Wrapping up this thread: I've [whitelisted specific error codes](https://github.com/woocommerce/woocommerce-gutenberg-products-block/pull/2851/files#diff-e82bdb511fe4e8a787289e47fef92f88R366) returned from `wc_create_new_customer()`, and pass the error message down for the error notice. I'm using `400` for all these errors, because they all represent client errors from the user's input, e.g. using an email that already has an account.